src/crypto/index.ts

```diff
@@ -1864,8 +1865,23 @@ export class Crypto extends EventEmitter {
                 }
 
                 if (this.getNeedsNewFallback()) {
-                    logger.info("generating fallback key");
-                    await this.olmDevice.generateFallbackKey();
+                    const fallbackKeys = await this.olmDevice.getFallbackKey();
+                    // if fallbackKeys is non-empty, we've already generated a
+                    // fallback key, but it hasn't been published yet, so we
+                    // can use that instead of generating a new one
+                    if (!fallbackKeys.curve25519 ||
+                        [...Object.keys(fallbackKeys.curve25519)].length == 0) {
```

Comment:
```suggestion
                        Object.keys(fallbackKeys.curve25519).length == 0) {
```

Reply:
Huh. I was under the impression that `Object.keys` returns an iterator rather than an array, but looks like I was wrong.